component/org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/mutualtls/MutualTLSClientAuthenticator.java

```diff
@@ -286,6 +292,17 @@ private X509Certificate parseCertificate(String content) throws CertificateExcep
                 .generateCertificate(new ByteArrayInputStream(decoded));
     }
 
+    /**
+     * Check if the string contains percent-encoded sequences.
+     *
+     * @param str input string value.
+     * @return returns true if string is url encoded.
+     */
+    private boolean isUrlEncoded(String str) {
```

Comment:
This is not a URL encode, but a certificate encode.

Reply:
Certificate is url encoded and we use `URLDecoder.decode` to decode

Reply:
From the gateway when appending the cert as a header, it does a url encoding. That needs to be decoded prior to the certificate decoding. That's what Anju is trying to do here.